File: frontend/src/types/safety-zone.ts
import { GeoJSONValue } from './robot-cell';
import { ZoneState, ZoneType } from './enums/zone-type';

export interface SafetyZone {
  id: number;
  robot_cell_id: number;
  robot_cell_code: string;
  name: string;
  zone_type: ZoneType;
  polygon_geojson: GeoJSONValue;
  min_height_mm: number;
  max_height_mm: number;
  speed_limit_mm_s: number;
  access_rule: string;
  zone_state: ZoneState;
  version: number;
  created_at: string;
  updated_at: string;
}

export interface CreateSafetyZone {
  robot_cell_id: number;
  name: string;
  zone_type: ZoneType;
  polygon_geojson: GeoJSONValue;
  min_height_mm: number;
  max_height_mm: number;
  speed_limit_mm_s: number;
  access_rule: string;
}

export interface UpdateSafetyZone extends Omit<CreateSafetyZone, 'robot_cell_id'> {
  version: number;
}

File: backend/internal/model/safety_zone.go
package model

import "time"

type SafetyZone struct {
	ID             uint      `gorm:"primaryKey"`
	RobotCellID    uint      `gorm:"index:idx_zone_cell_name,unique;not null"`
	Name           string    `gorm:"size:120;index:idx_zone_cell_name,unique;not null"`
	ZoneType       string    `gorm:"size:24;index;not null"`
	PolygonGeoJSON string    `gorm:"type:text;not null"`
	MinHeightMM    float64   `gorm:"not null"`
	MaxHeightMM    float64   `gorm:"not null"`
	SpeedLimitMMS  float64   `gorm:"column:speed_limit_mm_s;not null"`
	AccessRule     string    `gorm:"size:300;not null"`
	ZoneState      string    `gorm:"size:24;index;not null"`
	Version        int       `gorm:"not null;default:1"`
	CreatedBy      uint      `gorm:"index;not null"`
	CreatedAt      time.Time `gorm:"not null"`
	UpdatedAt      time.Time `gorm:"not null"`
	RobotCell      RobotCell `gorm:"foreignKey:RobotCellID"`
}
